aten/src/ATen/CMakeLists.txt

```diff
@@ -144,7 +144,7 @@ FILE(GLOB_RECURSE all_templates "templates/*")
 
 FILE(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/ATen)
 
-ADD_CUSTOM_COMMAND(OUTPUT ${generated_cpp}
+ADD_CUSTOM_COMMAND(OUTPUT ${generated_cpp} #need changes here
```

Comment:
Is there any way to make this add_custom_command executed before the build of `NativeFunctionsCuda.cu`?
The wrong build sequence has broken Windows CUDA builds.
@zdevito

Reply:
@colesbury was running into the same issue. I wonder if cmake has some kind of barrier, or a way to encode dependency for custom commands and their build artifacts

Reply:
@soumith I think this [post](https://samthursfield.wordpress.com/2015/11/21/cmake-dependencies-between-targets-and-files-and-custom-commands/) sheds some light on how to do this, but i don't exactly know that.

Reply:
We have a bunch of files (like all .cpp native functions) that depend on generated headers being built first to work. Why is this one any different? Could it be a cuda thing?

Reply:
Yes, it's related to CUDA. It seems that CMake is not working to getting the header files in `.cu` files generated first.

Reply:
it might be possible to add a specific dependency in this case, but it seems like a bug in CMake's cuda implementation.

Reply:
Yes, we can report this to CMake. But as a temporary workground, how could we add this dependency?